homeassistant/components/influxdb/__init__.py

```diff
@@ -319,7 +328,7 @@ def get_influx_connection(conf, test_write=False, test_read=False):
         def write_v2(json):
             """Write data to V2 influx."""
             try:
-                write_api.write(bucket=bucket, record=json)
+                write_api.write(bucket=bucket, record=json, write_precision=precision)
```

Comment:
If a user did not define precision, DEFAULT precision is used which differs from InfluxDB default precision. It would be nicer to check if precision is not None and if so, don't use the `write_precision` parameter.